broadlink/climate.py

```diff
@@ -238,3 +240,307 @@ def set_schedule(self, weekday: List[dict], weekend: List[dict]) -> None:
+
+        ambient_temp = payload[0x11] & 0b00011111
+        if ambient_temp:
+            data['ambient_temp'] = (ambient_temp
+                                    + float(payload[0x21] & 0b00011111) / 10.0)
+
+        checksum = self._calculate_checksum(payload[2:0x23]).to_bytes(2, 'big')
+        if (payload[0x23:0x25] != checksum):
+            logging.warning("checksum fail: calculated %s actual %s",
+                            checksum.hex(), payload[0x23:0x25].hex())
+
+        logging.debug("Received payload:\n%s", payload.hex(' '))
+
+        return data
+
+    def set_state(self, args: dict) -> bool:
+        """Set parameters of unit.
+
+        Args:
+            args (dict): if any are missing the current value will be retrived
+                state (bool): power
+                target_temp (float): temperature set point 16<n<32
+                mode (sq1.Mode): COOLING, HEATING, FAN, DRY, AUTO
+                speed (sq1.Speed): LOW, MID, HIGH, AUTO
+                mute (bool):
+                turbo (bool):
+                swing_h (sq1.Swing_H): ON, OFF
+                swing_v (sq1.Swing_V): ON, OFF, POS_1, POS_2, POS_3, POS_4, POS_5
+                sleep (bool)
+                display (bool)
+                health (bool)
+                clean (bool)
+                mildew (bool)
+
+        Returns:
+            True for success, verified by the unit's response.
+        """
+        CMND_0B_RMASK = 0b100
+        CMND_0C_RMASK = 0b1101
+        CMND_16 = 0b101
+
+        keys = ['state', 'mode', 'target_temp', 'speed', 'mute', 'turbo',
+                'swing_v', 'swing_h', 'sleep', 'display', 'health', 'clean',
+                'mildew']
+        unknown_keys = [key for key in args.keys() if key not in keys]
+        if len(unknown_keys) > 0:
+            raise ValueError(f"unknown argument(s) {unknown_keys}")
+
+        missing_keys = [key for key in keys if key not in args]
+        if len(missing_keys) > 0:
+            try:
+                received_state = self.get_state()
+            except RuntimeError as e:
+                if "unexpected payload size: 48" in str(e):
+                    # Occasionally a 48 byte payload gets mixed in,
+                    # a retry should suffice.
+                    received_state = self.get_state()
+                else:
+                    raise e
+            logging.debug("Raw args %s", args)
+            received_state.update(args)
+            args = received_state
+            logging.debug("Filled args %s", args)
+
+        args['target_temp'] = round(args['target_temp'] * 2) / 2
+        if not (args['target_temp'] >= 16 and args['target_temp'] <= 32):
+            raise ValueError(f"target_temp out of range, value: {args['target_temp']}")  # noqa E501
+
+        # Creating a new instance verifies the type
+        swing_R = self.Swing_H(args['swing_h']).value
+        swing_L = self.Swing_V(args['swing_v']).value
+
+        mode = self.Mode(args['mode']).value
```

Comment:
We can remove `.value` here. The result will always be an integer (with a cool interface).